connection.go

```diff
@@ -731,7 +728,7 @@ func (conn *Connection) reconnectImpl(neterr error, c Conn) {
 	if conn.opts.Reconnect > 0 {
 		if c == conn.c {
 			conn.closeConnection(neterr, false)
-			if err := conn.createConnection(true); err != nil {
+			if err := conn.createConnection(context.Background(), true); err != nil {
```

Comment:
We could use a context with timeout here.

Reply:
Updated.